src/lib.rs

```diff
@@ -86,3 +89,43 @@ impl TransmutingByteSlices for [u8] {
         unsafe { self.align_to::<u32>() }
     }
 }
+
+// Provide `[u32; 4]` and `[u64; 4]` iteration support for unaligned buffer.
+struct UnalignedBuffer<'a, T> {
```

Comment:
The old "numeric streams" kept a current pointer and an end pointer. WDYT about doing something similar:

```rust
    current: *const T,
    end: *const u8,
    phantom: PhantomData<&'a [u8]>,
}
```

Then, we can increment `current`, compare to `end`, and immediately dereference it if we are in bounds.

Reply:
That's a good idea. I was thinking about removing `offset` and updating `buf` in-place. The Rust fat pointer uses `size` instead of `end` so it's still slower than using pointers directly. `phantom` can be removed in this case, since the lifetime is on the struct directly.

Reply:
Seems reasonable!